pandas/_libs/tslibs/nattype.pyi

```diff
@@ -1,18 +1,12 @@
-from datetime import (
-    datetime,
-    timedelta,
-    tzinfo as _tzinfo,
-)
+from datetime import tzinfo as _tzinfo
 
 import numpy as np
 
-from pandas._libs.tslibs.period import Period
-
 NaT: NaTType
 iNaT: int
 nat_strings: set[str]
 
-_NaTComparisonTypes = datetime | timedelta | Period | np.datetime64 | np.timedelta64
+_NaTComparisonTypes = ...
```

Comment:
🤔 is this from `ruff`? looks a bit odd, will look into it

Reply:
Might need to declare it as a TypeAlias explicitly. `_NaTComparisonTypes: TypeAlias = datetime | ...`

Reply:
https://github.com/charliermarsh/ruff/issues/3860 will hopefully take care of at least the TypeVar cases. But we might need to add `TypeAlias` to non-TypeVar cases.